auto_submit/lib/validations/autosubmit_label_existence.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:auto_submit/model/auto_submit_query_result.dart';
+import 'package:auto_submit/validations/validation.dart';
+import 'package:github/github.dart' as github;
+
+import '../service/config.dart';
+
+/// Validates that the list of checks for the PR is not empty.
+class AutosubmitLabelExistence extends Validation {
+  AutosubmitLabelExistence({
+    required Config config,
+  }) : super(config: config);
+
+  @override
+
+  /// The `autosubmit` label must exist when landing PRs.
+  Future<ValidationResult> validate(QueryResult result, github.PullRequest messagePullRequest) async {
+    // List of labels associated with the pull request.
+    final List<String> labelNames = (messagePullRequest.labels as List<github.IssueLabel>)
+        .map<String>((github.IssueLabel labelMap) => labelMap.name)
+        .toList();
+    String message = '- The autosubmit label has been removed. Please add it back when ready.';
```

Comment:
Should we just log server side and do not add the comment to the PR? The message may be misleading specially because the label was removed by the author manually.

Reply:
Rethinking this, I thing `shouldProcess` is a better place for this label check. Updated.